pool/static_pool/pool.go

```diff
@@ -87,7 +87,7 @@ func NewPool(ctx context.Context, cmd pool.Command, factory pool.Factory, cfg *p
 	}
 
 	// set up workers allocator
-	p.allocator = pool.NewPoolAllocator(ctx, p.cfg.AllocateTimeout, factory, cmd, p.cfg.Command, p.log)
+	p.allocator = pool.NewPoolAllocator(ctx, p.cfg, factory, cmd, p.log)
```

Comment:
There is the reason, not to pass the whole configuration by pointer. Allocator just doesn't need these all configuration options. Furthermore, passing a pointer (rather than copy) allows you to change the values in the function accidentally during the initialization process.
Unlike Rust, in Golang, you can't pass a const reference, to forbid changes of the values passed by a reference.